pkgs/development/python-modules/gdsfactory/default.nix

```diff
@@ -0,0 +1,103 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  pytestCheckHook,
+  flit-core,
+  jinja2,
+  loguru,
+  matplotlib,
+  numpy,
+  orjson,
+  pandas,
+  pydantic,
+  pydantic-settings,
+  pydantic-extra-types,
+  pyyaml,
+  qrcode,
+  rectpack,
+  rich,
+  scipy,
+  shapely,
+  toolz,
+  types-pyyaml,
+  typer,
+  kfactory,
+  watchdog,
+  freetype-py,
+  mapbox-earcut,
+  networkx,
+  scikit-image,
+  trimesh,
+  ipykernel,
+  attrs,
+  graphviz,
+  # tests
+  jsondiff,
+  jsonschema,
+  pytest-regressions,
+}:
+buildPythonPackage rec {
+  pname = "gdsfactory";
+  version = "8.18.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "gdsfactory";
+    repo = "gdsfactory";
+    rev = "v${version}";
+    hash = "sha256-wDz8QpRgu40FB8+otnGsHVn2e6/SWXIZgA1aeMqMhPQ=";
+  };
+
+  nativeBuildInputs = [ flit-core ];
```

Comment:
```suggestion
  build-system = [ flit-core ];
```

Reply:
OK. I'll add all suggestions that I close and push them. Thanks for the review!